docs/developer.md

```diff
@@ -0,0 +1,312 @@
+---
+title: Developer Guide
+short_title: Developer Guide
+---
+
+## Developer quickstart
+
+These sections help you get started with a development environment for `mystmd` and learn how to contribute to the codebase. Note that `mystmd` is written in [TypeScript](https://www.typescriptlang.org/).
+
+We start by discussing the project architecture, and give a birds eye view of how the various parts fit together.
+Then, we delve into the specific tools you need, and the developer workflow for each part.
+Finally, we go over requirements for contributing code back, via GitHub Pull Requests.
+
+
+(architecture)=
+### Architecture
+
+From an author's perspective, `mystmd` can be thought of as a tool which compiles files in [MyST Markdown format](./quickstart-myst-markdown),
+a variant of Markdown with several extensions, into books, articles, and websites.
+Like other such tools, it can produce static output, such as PDF, docx, or HTML.
+
+However, for a developer it is important to understand that `mystmd` consists of two parts:
+
+1. An engine that converts input documents into an AST (abstract syntax tree).
+2. A renderer, that converts that AST into a given output format.
+
+This model is as applicable to static document conversion:
+
+```{mermaid}
+flowchart LR
+  subgraph "<tt>myst build</tt>"
+  myfile.md --> AST --> PDF
+  end
+```
+
+As it is to producing a rich web application:
+
+```{mermaid}
+flowchart TB
+  subgraph "<tt>myst start --headless</tt>"
+    subgraph <b>Document conversion</b>
+    direction LR
+    doc[folder_of/myfile.md] --> ast[AST]
+    end
+    subgraph <b>Content server</b>
+    ast --> content_server[serve AST on http://localhost:3100]
+    end
+  end
+  subgraph "<tt>npm run theme:book</tt>"
+    subgraph <b>Theme server</b>
+    direction LR
+    theme_server[render themed content at http://localhost:3000] --> content_server
+    user[User browser] --> theme_server
+    end
+  end
+```
+
+Walking through this last example, we see that invoking `myst start --headless` converts a set of input files into an AST, and serves up that AST via a content server.
+The theme server is a React app that knows how to style data, which it pulls in from the content server.
+The user interacts with the React app, which may trigger new fetches from the content server.
+
+#### Project-specific concepts
+
+`mystmd` is built on top of well known tools in the JavaScript ecosystem, such as A, B, and C for the `myst` CLI or [React](https://reactjs.org/), [Remix](https://remix.run/), and [Tailwind CSS](https://tailwindcss.com/) for the theme server.
```

Comment:
Unless we can add this content quickly, I suggest we just leave directives like the following and clean them up in future PRs

````
:::{error} to do - document myst CLI toolchain
:::
````